### ISSUE TITLE
False positives for 'no-duplicate-properties' and 'declarations-before-nesting' rules

### ISSUE THREAD

**Issue by chrismbarr** [sasstools/sass-lint#935]
Using `sass-lint` **1.10.0** (via `gulp-sass-lint` 1.3.0), both of the follow code blocks report a violation from the `no-duplicate-properties` rule where I've indicated with a comment.  Both of these lines are variable declarations and not even properties, so I have no idea why it's reporting these at all!

Luckily this release now includes rule disable comments, so I'm able to ignore these for now until this issue is resolved.

```scss
@mixin make-dashboard-grid-columns-float($columns, $class) {
    $list: '';

    @for $i from 1 to $columns {
        $list: '.col-#{$class}-#{$i}, #{$list}';
    }

    $i: $columns;
    $list: '#{$list}, .col-#{$class}-#{$i}';  // <--- lint error reported on this line

    #{$list} {
        // Inner gutter via padding
        padding-left: ($dashboard-grid-gutter / 2);
        padding-right: ($dashboard-grid-gutter / 2);
    }
}
```

```scss
.alert-tab {
    $subnav-color-alert-base: #921D1D;
    $subnav-color-alert-color: #FFECEC;
    $subnav-color-alert-bg: #B86B6B;
    $subnav-color-alert-bg-hover: #CD9898;
    $subnav-color-alert-bg-hover-gradient-secondary: #B36161;
    $subnav-color-alert-color-hover: #FFF;
    $subnav-color-alert-bg-hover: #FFF;  // <-- lint error reported on this line

    @include gradient-vertical($subnav-color-alert-bg, $subnav-color-alert-base, 30%, 85%);
    color: $subnav-color-alert-color;
    font-weight: bold;
    background-color: $subnav-color-alert-bg;

    &:hover {
        @include gradient-vertical($subnav-color-alert-bg-hover, $subnav-color-alert-bg-hover-gradient-secondary, 30%, 85%);
        color: $subnav-color-alert-color-hover;
        background-color: $subnav-color-alert-bg-hover;
    }

    &.current {
        color: $subnav-color-alert-base;

        &:hover {
            background: $subnav-color-alert-bg-hover; //force this BG color back on to replace the gradient
        }
    }
}
```

**Comment by DanPurdy** [comment]
Hi @chrismbarr Yeah, I actually noticed this earlier too, now that I've got the gulp release out of the way, I'll look at getting a hotfix out for a few of these issues that have popped up.

**Comment by chrismbarr** [comment]
Similarly, I noticed more false positives in this example as well for the `declarations-before-nesting` rule.

In general, it seems to confuse variables and properties.  It seems to treat any text that ends with a `:` as a property.

```scss
@mixin sprite-images($context, $image-path, $vertical-offset, $list) {
    // sass-lint:disable-block declarations-before-nesting
    #{$context} {
        background-image: url('#{$image-path}?v5');
    }

    $i: 0;   // <-- violation reported here

    @each $item in $list {
        #{$context}#{$context}_#{$item} {
            background-position: 0 $i;
        }

        $i: $i - $vertical-offset;  // <-- violation reported here
    }
}`
```
